pytools/graph.py

```diff
@@ -43,29 +49,44 @@
 Type Variables Used
 -------------------
 
-.. class:: T
+.. class:: NodeT
+
+    Type of a graph node, can be any hashable type.
+
+.. class:: GraphT
 
-    Any type.
+    A :class:`collections.abc.Mapping` representing a directed
+    graph. The mapping contains one key representing each node in the
+    graph, and this key maps to a :class:`collections.abc.Collection` of its
+    successor nodes. Note that most functions expect that every graph node
+    is included as a key in the graph.
 """
 
-from typing import (Collection, TypeVar, Mapping, List, Optional, Any,
-                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet)
+from typing import (Collection, Mapping, List, Optional, Any,
+                    Callable, Set, MutableSet, Dict, Iterator, Tuple,
+                    Hashable)
 
+try:
+    from typing import TypeAlias
+except ImportError:
+    from typing_extensions import TypeAlias
 
-T = TypeVar("T")
+
+NodeT: TypeAlias = Hashable
```

Comment:
So I missed that the node type went from a type variable to just `Hashable`, which is essential, and without which type checking in a bunch of downstream packages, e.g. here:

https://gitlab.tiker.net/inducer/pytato/-/jobs/498112

The correct thing would have been for `NodeT` to remain a type variable, and for `GraphT` to become a parametric type alias. I will push a fix in a bit.